### DISCUSSION TITLE
STAY AWAY from Gosund plugs

### DISCUSSION THREAD

**Discussion by samip5** [ct-Open-Source/tuya-convert#1123] [General]
I got confirmation from their support that most of their plugs are already upgraded versions and cannot be downgraded.

For reference:
```
From: Mufi (Support) <globalservice@gosund.com>
To: Me <censored>@gmail.com
Subject: Re: Firmware request

Hi Skyler,
Thank you for your email.
Most of our plugs are already upgraded and they no longer support flashing to Tasmota.
We are sorry to inform you that upon checking, your smart plug is an upgraded version. Therefore, the firmware cannot be updated to v1.0.8.
We are hoping for your understanding.
```

**Comment by mlinko6** [comment]
Which Version of Tasmota do you have?

KR
Rok

**Comment by samip5** [comment]
> Which Version of Tasmota do you have?
> 
> KR
> Rok

None? It comes with Tuya.

**Comment by mlinko6** [comment]
Sorry I mean’t Gosund version
________________________________

**Comment by samip5** [comment]
> Sorry I mean’t Gosund version

Main Module lists version as v1.1.15 and Other type: v1.0.7 so expect every single one of these to be non-downgradeable if you're currently thinking about them.

**Comment by mlinko6** [comment]
I've just flashed Gosund EP2 smart plugs with the firmware version: 1.0.8 4 with Tuya converter installed on Raspberry Pi4 - i had no Issues with flashing.

**Comment by samip5** [comment]
> I've just flashed Gosund EP2 smart plugs with the firmware version: 1.0.8 4 with Tuya converter installed on Raspberry Pi4 - i had no Issues with flashing.

That version is exactly the ONLY ONE which works, but most ones you can buy anymore are running newer than that, and thus will not work.

**Comment by broyuken** [comment]
you should still be able to open them up and flash them

**Comment by samip5** [comment]
> you should still be able to open them up and flash them

Doesn't matter, as I'm not going to open it.

**Comment by TrustedFloppa** [comment]
if you get some old models from the store then you may have 1.0.4 firmware.(Probably tuya forced them to stop flashing EP2 to customers plugs) But there is also EP2 with a beken chip. (EP2 RTL as gosund support says)
~~gosund ran out of esp chips??~~

**Comment by TrustedFloppa** [comment]
so sad they are patched, i didn't get to flash tasmota on mine :(

**Comment by Morphy99** [comment]
Thanks for the heads up. I have a few old gosuns kicking around which work and too many Sonoffs. Out of interest, does anyone recommend a nice quality smart plug for the UK with tuya-convert compatibility still?

**Comment by quentinsf** [comment]
Well, to my surprise and great delight, I have just managed to flash two Gosund UP111 plugs that I bought from Amazon last week (July 2024).  So do not give up all hope!

Notes from my experiences:
* I spent a long time trying but completely failed when using a Pi Zero W.  It seldom managed to generate the vtrust-flash SSID at all, and when it did, I couldn't connect to it.   I then switched to a Pi 4. (Temporarily re-purposing my Octoprint Pi is so easy when I just have to switch SD cards :-) )
* I was using the standard Raspberry Pi OS Lite, which is based on Debian Bookworm and installs Python 3.11.  When you try to run `install_prereq.sh`, it may refuse, saying that it's an externally-managed installation... (They try to encourage you do do everything in virtualenvs or containers or whatever.)  If, however, you just want to go ahead and install the python dependencies alongside the system site packages in the traditional way, run `sudo rm /usr/lib/python3.11/EXTERNALLY-MANAGED` before starting.

Then I basically just followed the instructions, noting that, as ever, it's _very_ important to make sure you successfully connect a phone or something else to the `vtrust-flash` network before trying the flash procedure, and that, if at first you don't succeed and need to retry, you may very well need to do this again.

Anyway, your mileage may vary, but I'm a happy bunny - and thanks for all those behind this very useful project!

**Reply by TrustedFloppa** [reply]
Well, that may be old manufacturing date like before updated firmware, so in this case you were mostly lucky lol
